weave-js/src/components/PagePanelComponents/Home/Browse2/Browse2OpDefCode.tsx

```diff
@@ -37,10 +37,29 @@ export const Browse2OpDefCode: FC<{uri: string; maxRowsInView?: number}> = ({
     );
   }
 
+  const detectLanguage = (code: string) => {
```

Comment:
For the purposes of unit testing and not recreating the function on each render I think it would be a little cleaner to pull this out out the class and just have it as a top-level function in the file that also takes uri as an argument.